### PR TITLE
ZCS-1219 :- Universal UI - Add filters modal, visual update

### PR DESCRIPTION
**Pull request by SatyamKumarOnline**
Desc :- Filter Rule dialog related changes. Code changes to match the layout changes as per mock .

Changeset :-
 WebRoot/css/zm.css
 WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js
 WebRoot/messages/ZmMsg.properties
 WebRoot/skins/_base/base4/skin.properties
 WebRoot/templates/prefs/Pages.template

### CODE REVIEW

File: WebRoot/css/zm.css [hunk 1]
```diff
@@ -4109,40 +4109,80 @@ HTML>BODY .appt-selected .appt_allday_body
...
+
+.ZmFilterRuleDialog .filterActions>tbody>tr> td:first-of-type {
+	@FilterRuleDialogActionSectionSelectWidget@
 }
 
 #IFDEF MSIE
 	/* IE won't pick up padding on the table cell elements unless the collapse is turned off */
-.FilterAddRemoveButtons {
-	border-collapse:separate;
+.ZmFilterRuleDialog .ZmFilterRemoveButtons {
```
Comment (kalpeshhpatel) [inline WebRoot/css/zm.css]: Do we need this rule only for IE? Have we verified that other browsers don't need it?

File: WebRoot/css/zm.css [hunk 2]
```diff
@@ -4109,40 +4109,80 @@ HTML>BODY .appt-selected .appt_allday_body
...
+}
+
+.ZmFilterRuleDialog .ZmFilterNameInput {
+	@FilterRuleDialogInputWrapper@
+}
+
+.ZmFilterRuleDialog .ZmFilterRuleSection,
+.ZmFilterRuleDialog .ZmFilterActionSection {
+	@FilterRuleDialogFieldSet@
+	#IFDEF MSIE
```
Comment (kalpeshhpatel) [inline WebRoot/css/zm.css]: I am not sure, but do we need this now? Have we tested on IE, if it's really needed?

File: WebRoot/templates/prefs/Pages.template [hunk 3]
```diff
@@ -1406,35 +1406,62 @@ NOTE: The tables inside of the filter rule sections MUST contain a tbody element
...
+							<table role="presentation" class='ZCheckboxTable'>
+								<tr>
+									<td><input id='${id}_active' type='checkbox' name='cbActive' checked tabindex=0></td>
+									<td class='ZmFilterActiveLabel'><label for='${id}_active'> <$=ZmMsg.active$> </label></td>
+								</tr>
+							</table>
+						</td>
+					</tr>
+					<tr>
+						<td class='ZmFilterNameInput' colspan='2'><input id='${id}_name' type='text' tabindex=0 style='width:100%;'></td>
```
Comment (kalpeshhpatel) [inline WebRoot/templates/prefs/Pages.template]: Avoid inline style

File: WebRoot/skins/_base/base4/skin.properties [hunk 4]
```diff
@@ -1681,6 +1682,27 @@ SmallOutsetBorder           = padding:1px; border:1px solid; border-color:@darke
 SmallInsetBorder            = padding:1px; border:1px solid; border-color:@darken(AppC,50)@ @darken(AppC,20)@ @darken(AppC,20)@ @darken(AppC,50)@;
 
 
+#######################################################################
+#   Filter Rule Dialog
+#######################################################################
+
+FilterRuleDialogOuterContainer                    = @DialogBackGroundColor@
+FilterRuleDialogFilterNameSection                 = margin-bottom:.75em; min-width:450px; margin-bottom:.75em;
```
Comment (kalpeshhpatel) [inline WebRoot/skins/_base/base4/skin.properties]: margin-bottom is repeated.

File: WebRoot/skins/_base/base4/skin.properties [hunk 5]
```diff
@@ -1681,6 +1682,27 @@ SmallOutsetBorder           = padding:1px; border:1px solid; border-color:@darke
...
+FilterRuleDialogFilterName                        = white-space:nowrap; padding-left:0px;
```
Comment (kalpeshhpatel) [inline WebRoot/skins/_base/base4/skin.properties]: You can use `NoWrap`

File: WebRoot/skins/_base/base4/skin.properties [hunk 6]
```diff
@@ -1681,6 +1682,27 @@ SmallOutsetBorder           = padding:1px; border:1px solid; border-color:@darke
...
+FilterRuleDialogCheckBoxLabel                     = white-space:nowrap; padding-left:@SkinViewInnerSpacing@;
```
Comment (kalpeshhpatel) [inline WebRoot/skins/_base/base4/skin.properties]: You can use `NoWrap`

File: WebRoot/css/zm.css (hunk 1 above)
Comment (SatyamKumarOnline) [inline WebRoot/css/zm.css]: I tested this on FF, briefly.
It's working fine.
This rule was earlier defined in the css file, I moved them to property file

File: WebRoot/css/zm.css (hunk 2 above)
Comment (SatyamKumarOnline) [inline WebRoot/css/zm.css]: This rule was earlier defined in the css file .

File: WebRoot/templates/prefs/Pages.template (hunk 3 above)
Comment (SatyamKumarOnline) [inline WebRoot/templates/prefs/Pages.template]: Done.
https://github.com/Zimbra/zm-web-client/pull/109/commits/007234ddf58b312fcf393a2428bb1e8669c7ac8d

File: WebRoot/skins/_base/base4/skin.properties (hunk 4 above)
Comment (SatyamKumarOnline) [inline WebRoot/skins/_base/base4/skin.properties]: Done.
https://github.com/Zimbra/zm-web-client/pull/109/commits/007234ddf58b312fcf393a2428bb1e8669c7ac8d

File: WebRoot/skins/_base/base4/skin.properties (hunk 5 above)
Comment (SatyamKumarOnline) [inline WebRoot/skins/_base/base4/skin.properties]: Done.
https://github.com/Zimbra/zm-web-client/pull/109/commits/007234ddf58b312fcf393a2428bb1e8669c7ac8d

File: WebRoot/skins/_base/base4/skin.properties (hunk 6 above)
Comment (SatyamKumarOnline) [inline WebRoot/skins/_base/base4/skin.properties]: Done.
https://github.com/Zimbra/zm-web-client/pull/109/commits/007234ddf58b312fcf393a2428bb1e8669c7ac8d

File: WebRoot/css/zm.css [hunk 7]
```diff
@@ -4113,40 +4113,89 @@ HTML>BODY .appt-selected .appt_allday_body
...
+}
+
+.ZmFilterRuleDialog .ZmFilterNameInput INPUT[type="text"]{
+	@FilterRuleDialogFilterNameInputField@
+}
+
+.ZmFilterRuleDialog .ZmFilterRuleSection,
+.ZmFilterRuleDialog .ZmFilterActionSection {
+	@FilterRuleDialogFieldSet@
+	#IFDEF MSIE
```
Comment (tarangkhandelwal) [inline WebRoot/css/zm.css]: indentation with closing (#ENDIF)

**Pull request merged**